fs.c

```diff
@@ -502,6 +493,25 @@ int FS(_wstat64) (const wchar_t *path, struct __stat64 *buffer)
   return result;
 }
 
+int FS(_wrename) (const wchar_t *from, const wchar_t *to)
+{
+  if (MoveFileW(from, to) == 0) {
```

Comment:
Almost, you need to call `create_device_name` on both `to` and `from`, which will convert the path between a normal Windows path into the win32 namespace format so the restrictions get removed.